File: scrfd/schemas.py
from pydantic import BaseModel


class Threshold(BaseModel):
    nms: float = 0.5
    probability: float = 0.4


class Point(BaseModel):
    x: float
    y: float


class Bbox(BaseModel):
    upper_left: Point
    lower_right: Point

    def height(self) -> float:
        return self.lower_right.y - self.upper_left.y

    def width(self) -> float:
        return self.lower_right.x - self.upper_left.x


class FaceKeypoints(BaseModel):
    left_eye: Point
    right_eye: Point
    nose: Point
    left_mouth: Point
    right_mouth: Point


class Face(BaseModel):
    bbox: Bbox
    probability: float
    keypoints: FaceKeypoints

File: scrfd-rust/src/schemas.rs
#[derive(Debug, Clone, Copy)]
pub struct Point {
    pub x: f32,
    pub y: f32,
}

#[derive(Debug, Clone, Copy)]
pub struct Bbox {
    pub x: f32,
    pub y: f32,
    pub w: f32,
    pub h: f32,
}

#[derive(Debug, Clone, Copy)]
pub struct FaceKeypoints {
    pub left_eye: Point,
    pub right_eye: Point,
    pub nose: Point,
    pub left_mouth: Point,
    pub right_mouth: Point,
}

#[derive(Debug, Clone, Copy)]
pub struct Face {
    pub score: f32,
    pub keypoints: FaceKeypoints,
    pub bbox: Bbox,
}

#[derive(Debug, Clone, Copy)]
pub struct Threshold {
    pub score: f32,
    pub iou: f32,
}
